.ci/robot_framework/libs/TestUtils.py

```diff
@@ -15,6 +15,31 @@ class TestUtils:
     def __init__(self):
         self.driver = None
 
+    def get_machine_expectation(self, id, machine, wpeversion, type="text"):
+        # Define file paths to search in order
+        files = [
+                os.path.join("..", "..", "robot_framework", "expectations", wpeversion, f"{machine}.expectations"),
+                os.path.join("..", "..", "robot_framework", "expectations", f"{machine}.expectations"),
+                os.path.join("..", "..", "robot_framework", "expectations", wpeversion, "generic.expectations"),
+                os.path.join("..", "..", "robot_framework", "expectations", "generic.expectations")
+                ]
+
+        for f in files:
+            if os.path.exists(f):
```

Comment:
Suggestion to avoid one extra nest level:

```python
for f in files:
    if not os.path.exists(f):
        continue
    with open(f, "r", encoding="utf-8") as fp:
        ...
```